src/fdw/base.rs

```diff
@@ -94,6 +93,16 @@ pub trait BaseFdw {
 
         let mut sql = format!("SELECT {targets} FROM {schema_name}.{table_name}");
 
+        if !quals.is_empty() {
+            let mut formatter = DuckDbFormatter::new();
+            let where_clauses = quals
+                .iter()
+                .map(|x| x.deparse_with_fmt(&mut formatter))
+                .collect::<Vec<String>>()
+                .join(" and ");
```

Comment:
Instead of allocated Vec and then join. Can we try to use `.intersperse(" AND ".to_string()) .collect::<String>();`?

Reply:
I’ve just learned that. Thank you! However, it seems that it’s an unstable feature. I’m not sure if we can use unstable features, as it appears the [workflow](https://github.com/paradedb/pg_analytics/blob/73ddff2ea1c587b8c1c11a499cf550adb283ebf9/.github/workflows/test-pg_analytics.yml#L71) specifies stable Rust.

Reply:
Let's stick to stable rust, yes

Reply:
This can come later once the feature is stable